### PR TITLE
Device prompting counter-proposal

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-util.c [hunk 1]
```diff
@@ -4252,13 +4269,14 @@ main(int argc, char *argv[])
 	}
 
 	/* non-TTY consoles cannot answer questions */
-	if (!fu_util_setup_interactive_console(&error_console)) {
+	if (!fu_util_setup_interactive_console(&error_console) || priv->as_json) {
 		g_debug("failed to initialize interactive console: %s", error_console->message);
```
Comment (hughsie) [inline src/fu-util.c:4273]: error_console won't be set here

**Review by hughsie — APPROVED** [review]
I fixed up the last commit, if that looks okay to me, good to merge. Thanks!

**Comment by superm1** [comment]
yeap looks  good

**Comment by hughsie** [comment]
I've rolled in https://github.com/fwupd/fwupd/pull/4312 too

**Comment by superm1** [comment]
Yup looks good.

**Pull request merged**